sources/kafka_pipeline.py

```diff
@@ -0,0 +1,75 @@
+from typing import Any, Dict
+
+import confluent_kafka  # type: ignore
+import dlt
+
+try:
+    from .kafka import kafka_consumer  # type: ignore
+except ImportError:
+    from kafka import kafka_consumer
+
+
+def load_from_several_topics() -> None:
+    pipeline = dlt.pipeline(
+        pipeline_name="kafka_pipeline",
+        destination="postgres",
+        dataset_name="kafka_messages",
+    )
+
+    data = kafka_consumer(["purchases", "books"])
+
+    info = pipeline.run(data)
+    print(info)
+
+
+def load_with_custom_processor() -> None:
+    pipeline = dlt.pipeline(
+        pipeline_name="kafka_pipeline",
+        destination="postgres",
+        dataset_name="kafka_messages",
+    )
+
+    def custom_msg_processor(msg: confluent_kafka.Message) -> Dict[str, Any]:
+        """Convert the given message before loading into the destination.
+
+        Args:
+            msg (confluent_kafka.Message): Message read from the cluster.
+
+        Returns:
+            dict: converted message.
+        """
+        return {
+            "_kafka": {
+                "topic": msg.topic(),  # required field
+                "key": msg.key().decode("utf-8"),
```

Comment:
bare in mind that this doesn't work all the time, you might want to provide a way for the users to specific the type of encoding used.
e.g.
```
def message_key_decoder(*, decoder_type, message_key, deserializer):
    """
    Given the type of decoder it returns the decoded message as dict
    """
    if decoder_type == 'json' or decoder_type == 'json_schema':
        return message_key.decode('utf-8')
    elif decoder_type == 'avro_schema':
        return deserializer.__call__(message_key, None)
    else:
        raise ValueError('Decoder type not supported')
```

if the key is avro serialized this will fail.